### PR TITLE
Helper script for cloning framework content from a previous iteration

### PR DESCRIPTION
**Pull request by katstevens**
Automates some of the initial steps here: https://alphagov.github.io/digitalmarketplace-manual/content-and-frameworks/adding-frameworks.html#id2

It's verbose/non-DRY in places, to try and keep visibility of the different files being edited.

Tested with G-Cloud 12 and DOS 5, but not exhaustively.

Scope for more unit tests to be added at a later date 😸

### CODE REVIEW

File: script_helpers/clone_helpers.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    last_space_position = name.rindex(" ")
+    return name[:last_space_position] + "&nbsp;" + name[(last_space_position + 1):]
+
+
+class FrameworkContentCloner:
+
+    def __init__(self, framework_family, iteration_number, launch_year):
+        self._launch_year = launch_year
+        self._new_fw_slug = f"{framework_family}-{iteration_number}"
+        self._previous_fw_slug = f"{framework_family}-{iteration_number - 1}"
```
Comment (risicle) [inline script_helpers/clone_helpers.py]: Not expecting to have to care about e.g. DOS1 omitting the number altogether, are we?

File: scripts/clone-latest-framework.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+"""
+import os
+import sys
+sys.path.insert(0, '.')
+from docopt import docopt
+from script_helpers.clone_helpers import FrameworkContentCloner
+
+
+if __name__ == '__main__':
+    current_dir = os.path.dirname(__file__)
```
Comment (risicle) [inline scripts/clone-latest-framework.py]: Das ist nicht used, nein?

File: scripts/clone-latest-framework.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+- Update metadata
+- Update hardcoded content of framework name in declaration questions 'G-Cloud 11', 'G-Cloud&nsbp;11'
+- Update hardcoded content of framework slug in urls, declaration
+- Set any dates to a point in the far-future e.g. 1 Jan 2525
+
+Usage:
+    clone-latest-framework.py --family=<framework_family> --iteration=<iteration_number> --launch-year=<launch_year>
+
+Example:
+    ./scripts/clone-latest-framework.py --family=g-cloud --iteration=12 --launch-year=2020
```
Comment (risicle) [inline scripts/clone-latest-framework.py]: It's not been made clear here what `--launch-year` is for.

File: tests/test_clone_latest_framework.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    assert cloner._launch_year == 2020
+    assert cloner._new_fw_slug == 'g-cloud-12'
+    assert cloner._previous_fw_slug == 'g-cloud-11'
+    assert cloner._following_fw_slug == 'g-cloud-13'
+
+    assert cloner._previous_name == 'G-Cloud 11'
+    assert cloner._new_name == 'G-Cloud 12'
+    assert cloner._previous_nbsp_name == 'G-Cloud&nbsp;11'
+    assert cloner._escaped_previous_nbsp_name == 'G&#x2011;Cloud&nbsp;11'
+    assert cloner._new_nbsp_name == 'G-Cloud&nbsp;12'
```
Comment (risicle) [inline tests/test_clone_latest_framework.py]: Now a _purist_ would scream at you for testing a non-public interface.
(I'm not a purist though)

File: script_helpers/clone_helpers.py (hunk 1 above)
Comment (katstevens) [reply script_helpers/clone_helpers.py]: Good point!

File: tests/test_clone_latest_framework.py (hunk 4 above)
Comment (katstevens) [reply tests/test_clone_latest_framework.py]: Hah, there are barely any tests as it is - I'm not removing these 😸

File: script_helpers/clone_helpers.py (hunk 1 above)
Comment (katstevens) [reply script_helpers/clone_helpers.py]: Have added a commit to handle `iteration_number = 1` as well.

**Pull request merged**